azure/services/networkinterfaces/networkinterfaces.go

```diff
@@ -105,6 +111,12 @@ func (s *Service) Delete(ctx context.Context) error {
 	//  Order of precedence (highest -> lowest) is: error that is not an operationNotDoneError (i.e. error deleting) -> operationNotDoneError (i.e. deleting in progress) -> no error (i.e. deleted)
 	var result error
 	for _, nicSpec := range specs {
+		if managed, err := s.isManaged(ctx, nicSpec); err == nil && !managed {
+			log.V(4).Info("Skipping network interface deletion in custom network interface mode")
```

Comment:
This might be an opportunity to make logging more visible and add a link to https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3575/files#diff-a109f59250df65bd0d5487b914d0bce7b1a07717583e9414cfe2c5e11f4fcf13R45 for users that may not have read the release notes or docs